### ISSUE TITLE
[Bug]:  Incorrect results with ONNX and OpenVINO after training Patchcore with default settings – all predictions are abnormal

### ISSUE THREAD

**Issue by adminaccount001** [open-edge-platform/anomalib#2303]
### Describe the bug

I trained the Patchcore model using default settings.
In the PyTorch inference, the model can correctly distinguish between normal and abnormal samples.
However, after exporting the model to ONNX and OpenVINO, all inference results are classified as abnormal, meaning the model fails to correctly predict normal samples.

### Dataset

Folder

### Model

PatchCore

### Steps to reproduce the behavior

Train Patchcore:

```python
datamodule = Folder(
    name="crop1",
    root=r"E:\crop1",
    normal_dir="normal",
    abnormal_dir="abnormal",
    mask_dir="mask",       
    task=TaskType.SEGMENTATION,
    num_workers=0,
    image_size=[640,640],
    val_split_ratio=0.5,
    train_batch_size=1,
    eval_batch_size=1,
)

# Setup the datamodule
datamodule.setup()

model = Patchcore()
engine = Engine(
    task=TaskType.SEGMENTATION,
    )
engine.train(datamodule=datamodule, model=model)
```
Export
```python
engine.export(
    model= model,
    export_type=ExportType.OPENVINO,
    export_root=r'E:\results\Patchcore\crop1',
    input_size=[640,640],
    ckpt_path=r'E:\crop1\latest\weights\lightning\model.ckpt',
    )
```

I received the following warning during the export process:
![image](https://github.com/user-attachments/assets/b02470d6-45ac-4ce8-b310-a546237f4f85)

### OS information

OS information:
- OS: Windows 11
- Python version: 3.10.11
- Anomalib version: 1.1.0
- PyTorch version: 2.2.2cu118

### Expected behavior

Additionally, when I commented out the center crop operation in the Patchcore model, the issue still persisted. I was still unable to get correct results from ONNX or OpenVINO.

### Pip/GitHub

pip

### Configuration YAML

```yaml
N/A
```

### Logs

```shell
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by samet-akcay** [comment]
Which inference method do you use for Torch? `TorchInferencer` or `Enginer.predict`?
It could possibly be a normalization issue? @ashwinvaidya17, any thoughts on this?

**Comment by samet-akcay** [comment]
Thanks for sharing. @djdameln adding you here as well since you are working on OpenVINO inference refactor.

**Comment by samet-akcay** [comment]
@watertianyi, you don't need to use OpenVINO inference if you have NVIDIA GPU. You could just use Lightning or TorchInferencer, which should work fine. 

here, you could find more information about OpenVINO GPU support 
https://docs.openvino.ai/2024/notebooks/gpu-device-with-output.html

**Comment by watertianyi** [comment]
@samet-akcay 
The above problem has been solved. The main reason is that input_size cannot be set when converting OpenVINO because the error message is dynamic size. Is this correct? I would like to inquire whether OpenVINO supports NVIDIA GPU. The second question is if I want to improve the patchcore inference speed. What should be done? Because I have tried TorchInferencer and batchsize batch testing, the inference time is the same, and there is no change.

**Comment by Abhijeet241093** [comment]
engine.export(
    model= model,
    export_type=ExportType.OPENVINO,
    export_root=r'E:\results\Patchcore\crop1',
    input_size=[640,640],
    ckpt_path=r'E:\crop1\latest\weights\lightning\model.ckpt',
    ) When you define this, where is ExportType class:? in which packet it define ?

**Comment by GanAndreas** [comment]
> TorchInferencer. Testing revealed this issue is due to differences in how resize is implemented in OpenCV and PIL. It's recommended to replace cv2.resize() with PIL's Image.resize().
> 
> > [anomalib/src/anomalib/deploy/inferencers/openvino_inferencer.py](https://github.com/openvinotoolkit/anomalib/blob/cdd338c8992efab6032e7e29d0069ec3f13d5bbb/src/anomalib/deploy/inferencers/openvino_inferencer.py#L204)
> >     
> >     
> >          Line 204
> >       in
> >       [cdd338c](/openvinotoolkit/anomalib/commit/cdd338c8992efab6032e7e29d0069ec3f13d5bbb)
> >     
> >   
> >   
> >     
> > 
> >         
> >           
> >            image = cv2.resize(image, tuple(list(self.input_blob.shape)[2:][::-1]))

I fixed it by replacing the line with this
`cv2.resize(image, tuple([int(str(self.input_blob.partial_shape[2])), int(str(self.input_blob.partial_shape[3]))]))`

**Comment by samet-akcay** [comment]
@GanAndreas thanks for sharing your solution. 

Also, note that Anomalib v2 does not use cv2 anymore
https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/deploy/inferencers/openvino_inferencer.py
